fastchat/serve/gradio_block_arena_vision_anony.py

```diff
@@ -242,6 +246,56 @@ def clear_history(request: gr.Request):
         + [""]
     )
 
+def extract_text_from_pdf(pdf_file_path):
+    """Extract text from a PDF file."""
+    try:
+        with open(pdf_file_path, 'rb') as f:
+            reader = PyPDF2.PdfReader(f)
+            pdf_text = ""
+            for page in reader.pages:
+                pdf_text += page.extract_text()
+            return pdf_text
+    except Exception as e:
+        logger.error(f"Failed to extract text from PDF: {e}")
+        return None
+
+def llama_parse(pdf_path):
+    os.environ['LLAMA_CLOUD_API_KEY'] = 'LLAMA KEY'
+
+    output_dir = 'outputs'
+    os.makedirs(output_dir, exist_ok=True)
+
+    pdf_name = os.path.splitext(os.path.basename(pdf_path))[0]
+    markdown_file_path = os.path.join(output_dir, f'{pdf_name}.md')
+
+    command = [
+        'llama-parse',
+        pdf_path,
+        '--result-type', 'markdown',
+        '--output-file', markdown_file_path
+    ]
+
+    subprocess.run(command, check=True)
```

Comment:
Should we use their python SDK instead of invoking with `subprocess`?
https://github.com/run-llama/llama_parse?tab=readme-ov-file#using-with-file-object

Reply:
Yes, I changed it to parse the file object directly.